### DISCUSSION TITLE
mismatch betwee special tokens of the tokenizer and the model on qwen3.5

### DISCUSSION THREAD

**Discussion by ist1373** [QwenLM/Qwen3.6#96] [Q&A]
I am trying to fine-tune qwen 3.5 on a dataset. To make everything have set correctly I printed the model and tokenizer special tokens of qwen 3.5 and I got:

tokenizer.eos_token_id      = 248046
model.config.eos_token_id   = 248044
model.gen.eos_token_id      = [248046, 248044]
tokenizer.pad_token_id      = 248044
model.config.pad_token_id   = None
model.gen.pad_token_id      = 248044

as the model is looking at config.eos_token_id during the training (correct me if I am wrong pls) and the dataset is pretokenized using tokenizer.eos_token_id, is not that broken the training pipeline?

**Comment by KOKOSde** [comment]
This mismatch does not break training by itself. The token you append in the dataset is what the loss is trained on. `model.config.eos_token_id` is mostly a default, while `generation_config.eos_token_id` is what generation usually follows. Qwen includes both stop ids there, so inference can stop on either. If you want everything to look consistent, set `model.config.eos_token_id = tokenizer.eos_token_id` and `model.config.pad_token_id = tokenizer.pad_token_id` before training, but your pretokenized data is not invalid because of this mismatch.
